### PR TITLE
More robust initialization of u and v in `examples/two_dimensional_turbulence.jl`

### LINKED ISSUE #1656: Change of boundary condition from Periodic to Bounded problem
I run https://github.com/CliMA/Oceananigans.jl/blob/master/examples/two_dimensional_turbulence.jl with success

but when trying to adapt the  y-boundary ( or x or both) to Bounded

grid = RegularRectilinearGrid(topology=(Periodic, Bounded, Flat),size=(128, 128),extent=(2π, 2π))

 I get the following error (I included a show to show the grid and its size)

```julia
grid = RegularRectilinearGrid{Float64, Periodic, Bounded, Flat}
                   domain: x ∈ [0.0, 6.283185307179586], y ∈ [-1.2810265668750775e-18, 6.283185307179586], z ∈ [0.0, 0.0]
                 topology: (Periodic, Bounded, Flat)
  resolution (Nx, Ny, Nz): (128, 128, 1)
   halo size (Hx, Hy, Hz): (1, 1, 0)
grid spacing (Δx, Δy, Δz): (0.04908738521234052, 0.04908738521234052, 0.0)
┌ Warning: Inflating model grid halo size to (3, 3, 0) and recreating grid. The model grid will be different from the input grid. To avoid this warning, pass halo=(3, 3, 0) when constructing the grid.
└ @ Oceananigans.Grids C:\Users\jmbeckers\.julia\packages\Oceananigans\zj42o\src\Grids\automatic_halo_sizing.jl:41
size(model.grid) = (128, 128, 1)
ERROR: LoadError: DimensionMismatch("array could not be broadcast to match destination")
Stacktrace:
 [1] check_broadcast_shape at .\broadcast.jl:520 [inlined]
 [2] check_broadcast_shape at .\broadcast.jl:521 [inlined]
 [3] check_broadcast_axes at .\broadcast.jl:523 [inlined]
 [4] instantiate(::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{3},Tuple{Base.OneTo{Int64},Base.OneTo{Int64},Base.OneTo{Int64}},typeof(identity),Tuple{Array{Float64,3}}}) at .\broadcast.jl:269
 [5] materialize! at .\broadcast.jl:848 [inlined]
 [6] materialize! at C:\Users\jmbeckers\.julia\packages\Oceananigans\zj42o\src\Fields\broadcasting_abstract_fields.jl:19 [inlined]
 [7] set!(::Field{Center,Face,Center,CPU,OffsetArrays.OffsetArray{Float64,3,Array{Float64,3}},RegularRectilinearGrid{Float64,Periodic,Bounded,Flat,OffsetArrays.OffsetArray{Float64,1,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}},Float64,NamedTuple{(:x, :y, :z),Tuple{CoordinateBoundaryConditions{BoundaryCondition{Oceananigans.BoundaryConditions.Periodic,Nothing},BoundaryCondition{Oceananigans.BoundaryConditions.Periodic,Nothing}},CoordinateBoundaryConditions{BoundaryCondition{Oceananigans.BoundaryConditions.NormalFlow,Nothing},BoundaryCondition{Oceananigans.BoundaryConditions.NormalFlow,Nothing}},CoordinateBoundaryConditions{Nothing,Nothing}}}}, ::Array{Float64,3}) at C:\Users\jmbeckers\.julia\packages\Oceananigans\zj42o\src\Fields\set!.jl:16
 [8] set!(::IncompressibleModel{Oceananigans.TimeSteppers.RungeKutta3TimeStepper{Float64,NamedTuple{(:u, :v, :w),Tuple{Field{Face,Center,Center,CPU,OffsetArrays.OffsetArray{Float64,3,Array{Float64,3}},RegularRectilinearGrid{Float64,Periodic,Bounded,Flat,OffsetArrays.OffsetArray{Float64,1,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}},Float64,NamedTuple{(:x, :y, :z) ....
```

Julia 1.5.2 and sorry if I missed something obvious

### PR DESCRIPTION
**Pull request by glwagner**
This PR initializes `u` and `v` separately and in a way that's more robust to user changes to the script (for example, if users change the grid topology).

Resolves #1656 . Thanks @jmbeckers for pointing out this issue (apologies it took so long to resolve!)

### CODE REVIEW

**Review by francispoulin — APPROVED** [review]
Good idea, unfortunately a test fails.

**Comment by navidcy** [comment]
All CPU tests don't even run... It's been happening past few days. Same, e.g., in #1915..

File: examples/two_dimensional_turbulence.jl [hunk 1]
```diff
@@ -42,10 +42,15 @@ model = NonhydrostaticModel(timestepper = :RungeKutta3,
...
+u, v, w = model.velocities
+
+uᵢ = rand(size(u)...)
+vᵢ = rand(size(v)...)
+
+uᵢ .-= mean(uᵢ)
+vᵢ .-= mean(vᵢ)
 
-set!(model, u=u₀, v=u₀)
+set!(model, u=u₀, v=v₀)
```
Comment (tomchor) [inline examples/two_dimensional_turbulence.jl:53]:
```suggestion
set!(model, u=uᵢ, v=vᵢ)
```

Comment (tomchor) [reply examples/two_dimensional_turbulence.jl:53]: I think tests aren't passing because of this typo

**Pull request merged**